src/tools/wasm-ctor-eval.cpp

```diff
@@ -165,41 +168,92 @@ class EvallingModuleInstance
   }
 };
 
+// Build an artificial `env` module based on a module's imports, so that the
+// interpreter can use correct object instances. It initializes usable global
+// imports, and fills the rest with fake values since those are dangerous to
+// use. we will fail if dangerous globals are used.
+std::unique_ptr<Module> buildEnvModule(Module& wasm) {
+  auto env = std::make_unique<Module>();
+  env->name = "env";
+
+  // create empty functions with similar signature
+  ModuleUtils::iterImportedFunctions(wasm, [&](Function* func) {
+    if (func->module == "env") {
+      auto* copied = ModuleUtils::copyFunction(func, *env);
+      copied->module = Name();
+      copied->base = Name();
```

Comment:
A non-imported function has to have a body. I think we don't validate it, so this went missed.

The body can be an `Unreachable` which would guard against it somehow being called.

Reply:
Thanks for the explanation. @tlively had a similar comment. It's now resolved.